### PR TITLE
Dr down

### PR DESCRIPTION
**Pull request by christophsax**
``` r
indiedown::dr_down()
#> 
#> ── System Information ──────────────────────────────────────────────────────────
#> ℹ R-Version: 4.1.0
#> ℹ Operating System: darwin17.0
#> ✔ tinytex Version: 0.31
#> ✔ pandoc Version: 2.11.4
#> ℹ indiedown Version: 0.1.1
#> ℹ Running outside of RStudio
#> 
#> ── Test Runs ───────────────────────────────────────────────────────────────────
#> ✔ Running lualatex successfully
#> ✔ Running xelatex successfully
#> ✔ Running pdflatex successfully
```

<sup>Created on 2021-07-05 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

It would be quite useful to have that on CRAN, so we can run it on systems with limited GitHub access. The function will likely evlolve over time.

### CODE REVIEW

**Comment by krlmlr** [comment]
Naming suggestion: `sit_down()` (abbreviated from "situation report for cynkradown").

Detailed review follows.

File: R/dr_down.R [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+  if (!is.na(version_info['rstudio'])) {
+    cli_alert_success("RStudio: {version_info['rstudio']}")
+  } else {
+    cli_alert_info("Running outside of RStudio")
+  }
+
+
+  cli_h1("Test Runs")
+  text <- paste("Running", names(is_success), ifelse(is_success, "successfully", "unsucessfully"))
```
Comment (krlmlr) [inline R/dr_down.R:98]:
```suggestion
  text <- paste("Running", names(is_success), ifelse(is_success, "successfully", "unsuccessfully"))
```

File: R/dr_down.R [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  pandoc_v <- as.character(rmarkdown::pandoc_version())
+  r_v <- paste(version$major, version$minor, sep = ".")
+
+  if (Sys.getenv("RSTUDIO") == "1") {
+    rstudio_v <- as.character(rstudioapi::versionInfo()$version)
+  } else {
+    rstudio_v <- NA_character_
+  }
+
+  if("tinytex" %in% rownames(installed.packages())){
```
Comment (krlmlr) [inline R/dr_down.R:22]: `rlang::is_installed()` or `requireNamespace(quietly = TRUE)` is faster.

File: R/dr_down.R [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    tinytex_v <- as.character(packageVersion("tinytex"))
+    is_tinytex = tinytex::is_tinytex()
```
Comment (krlmlr) [inline R/dr_down.R:24]: Inconsistent use of `=` and `<-` here and elsewhere.

File: R/dr_down.R [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    if (is_success[i]) {
+      cli_alert_success(text[i])
+    } else {
+      cli_alert_danger(text[i])
+    }
+  }
+
+  ans <- list(
+    version_info = version_info,
+    is_success = is_success,
```
Comment (krlmlr) [inline R/dr_down.R:109]: The `is_` prefix is misleading, just drop it?

**Review by krlmlr — COMMENTED** [review]
Thanks, this is an important first aid tool.

**Comment by christophsax** [comment]
> Naming suggestion: `sit_down()` (abbreviated from "situation report for cynkradown").
> 

I feel that `dr_down()` is a million times more self explanatory and easier to remember than `sit_down()`.

File: R/dr_down.R [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+          rmarkdown::pdf_document(latex_engine = latex_engines),
+          quiet = TRUE
+        )
+      ),
+      error = identity
+    )
+  }
+
+  ans <- lapply(latex_engines, try_with_engine)
+  success <- setNames(!sapply(ans, inherits, "try-error"), latex_engines)
```
Comment (krlmlr) [inline R/dr_down.R:65]: With `tryCatch()` the class changes.

```suggestion
  success <- setNames(!sapply(ans, inherits, "error"), latex_engines)
```

File: R/dr_down.R [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  if (!is.na(version_info['rstudio'])) {
+    cli_alert_success("RStudio: {version_info['rstudio']}")
+  } else {
+    cli_alert_info("Running outside of RStudio")
+  }
+
+
+  cli_h1("Test Runs")
+  text <- paste("Running", names(success), ifelse(success, "successfully", "unsuccessfully"))
+  for (i in seq(text)) {
```
Comment (krlmlr) [inline R/dr_down.R:100]: I didn't even know this works. For clarity, I tend to use:

```suggestion
  for (i in seq_along(text)) {
```

**Review by krlmlr — APPROVED** [review]
Nice! We could get full test coverage by extracting the code that detects if software is installed into separate functions, and mocking them. An exercise left to the reader.

**Pull request merged**